### PR TITLE
Fix user entity serialize

### PR DESCRIPTION
**Pull request by lenonleite**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR is fixing this deprecation:
```
PHPUnit\Framework\Exception: PHP Deprecated:  Mautic\UserBundle\Entity\User implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in /home/runner/work/mautic/mautic/app/bundles/UserBundle/Entity/User.php on line 19
```

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

**Comment by escopecz** [comment]
@lenonleite I was looking into this a little and I can't find why the User entity even have to implement the `Serializable` interface and implement the methods. I don't see them being used. It was added in the very fist implementation of this class:

https://github.com/escopecz/mautic/commit/ddbba81bfd6e455d7f35d9f9d2825192ff9eaea0#diff-32e00ed938d78ee24064b9afececaa7c0a31ef066dfdf8baa929f63b3be8e892R194

Perhaps it's not needed anymore? It is a bit silly that the methods will now do the opposite then before. So I think it's safer to remove them and test that users can still log in/out.

Edit: It's being used by Symfony:

https://symfony.com/doc/current/security.html#understanding-how-users-are-refreshed-from-the-session

File: app/bundles/UserBundle/Entity/User.php [hunk 1]
```diff
@@ -393,32 +393,48 @@ public function eraseCredentials()
...
             $this->isPublished(),
-        ]);
+        ];
     }
 
     /**
      * {@inheritdoc}
      */
-    public function unserialize($serialized)
+    public function serialize()
```
Comment (escopecz) [inline app/bundles/UserBundle/Entity/User.php]: @lenonleite please remove the original `serialize()` and `unserialize()` methods as discussed in https://mautic.slack.com/archives/CQMKV0RU1/p1686064066207829

Comment (escopecz) [reply app/bundles/UserBundle/Entity/User.php]: It looks like we'll also have to remove the `Serializable` interface with those methods.

**Comment by escopecz** [comment]
Testing of this PR is currently blocked by https://github.com/mautic/mautic/pull/11867 - merged now 🎉

**Review by mabumusa1 — APPROVED** [review]
I did a code review, code changes looks good

**Review by escopecz — APPROVED** [review]
I tested this and I can log out, log in and browse the administration

**Pull request merged**